Zend/zend_builtin_functions.c

```diff
@@ -496,15 +494,14 @@ ZEND_FUNCTION(define)
 	ZEND_PARSE_PARAMETERS_END();
 
 	if (zend_memnstr(ZSTR_VAL(name), "::", sizeof("::") - 1, ZSTR_VAL(name) + ZSTR_LEN(name))) {
-		zend_error(E_WARNING, "Class constants cannot be defined or redefined");
-		RETURN_FALSE;
+		zend_argument_value_error(1, "cannot be a class constant");
+		RETURN_THROWS();
 	}
 
 	if (non_cs) {
-		zend_error(E_WARNING,
-			"define(): Declaration of case-insensitive constants is no longer supported");
-		RETURN_FALSE;
-	}
+		zend_error(E_WARNING, "define(): Declaration of case-insensitive constants is no longer supported");
+    	RETURN_FALSE;
```

Comment:
Not really related to your changes, but I'm wondering if it wouldn't make more sense to make this declare a case-sensitive constant, rather than return false. I think it should be either that, or throw.

Reply:
I went with the first solution